academy/ibc/ibc-tao-dev.md

```diff
@@ -0,0 +1,314 @@
+
+<HighlightBox type="info">
+
+**Version Negotiation**
+
+Note that versioning here refers to the IBC protocol spec and not the ibc-go module. A backwards incompatible update is currently not planned.
+
+```go
+// Version defines the versioning scheme used to negotiate the IBC verison in
+// the connection handshake.
+type Version struct {
+	// unique version identifier
+	Identifier string `protobuf:"bytes,1,opt,name=identifier,proto3" json:"identifier,omitempty"`
+	// list of features compatible with the specified identifier
+	Features []string `protobuf:"bytes,2,rep,name=features,proto3" json:"features,omitempty"`
+}
+```
+
+Protocol versioning is important to establish, as different protocol versions may not be compatible due to ie: proofs being stored on a different path. There are three types of protocol version negotiation:
+
+*Default, no selection* only one protocol version is supported, this is the default one to propose.
+
+*With selection* two protocol versions can be proposed, such that the chain initiating `OpenInit` or `OpenTry` has a choice of which version to go with.
+
+*Impossible communication* backwards incompatible IBC protocol version -- ie: if IBC module changes where it stores its proofs (proof paths), errors out. So far, there are no plans to upgrade to a backwards incompatible IBC protocol version.
+
+</HighlightBox>
+
+As discussed above, the opening handshake protocol allows each chain to verify the identifier used to reference the connection on the other chain, enabling modules on each chain to reason about the reference of the other chain.
+
+Take a look at the [connection Protobuf](https://github.com/cosmos/ibc-go/blob/main/proto/ibc/core/connection/v1/connection.proto) for the `ConnectionEnd`:
+
+```
+message ConnectionEnd {
+  option (gogoproto.goproto_getters) = false;
+  // client associated with this connection.
+  string client_id = 1 [(gogoproto.moretags) = "yaml:\"client_id\""];
+  // IBC version which can be utilised to determine encodings or protocols for
+  // channels or packets utilising this connection.
+  repeated Version versions = 2;
+  // current state of the connection end.
+  State state = 3;
+  // counterparty chain associated with this connection.
+  Counterparty counterparty = 4 [(gogoproto.nullable) = false];
+  // delay period that must pass before a consensus state can be used for
+  // packet-verification NOTE: delay period logic is only implemented by some
+  // clients.
+  uint64 delay_period = 5 [(gogoproto.moretags) = "yaml:\"delay_period\""];
+}
+```
+
+`ConnectionEnd` defines a stateful object on a chain connected to another separate one. Note that there must only be 2 defined `ConnectionEnd`s to establish a connection between two chains. So the connections are mapped and stored as `ConnectionEnd` on the chain.
```

Comment:
`ConnectionEnd` defines a stateful object on a chain connected to its counterpart on the counterparty chain. Note that there must only be 2 defined `ConnectionEnd`s to establish a connection between two chains. So the connections are mapped and stored as `ConnectionEnd` on the respective chains.